src/frontend/src/utils/stream_graph_formatter.rs

```diff
@@ -93,6 +135,87 @@ impl StreamGraphFormatter {
             _ => node.identity.clone(),
         };
         writeln!(f, "{}{}", " ".repeat(level * 2), one_line_explain)?;
+        let explain_table_oneline = match node.get_node_body().unwrap() {
+            stream_node::NodeBody::Materialize(node) => Some(format!(
+                "materialized table: {}",
+                self.add_table(node.get_table().unwrap())
+            )),
+            stream_node::NodeBody::GlobalSimpleAgg(node) => Some(format!(
+                "state tables: [{}]",
+                node.internal_tables
+                    .iter()
+                    .map(|tb| { self.add_table(tb) })
+                    .join(", ")
+            )),
+            stream_node::NodeBody::HashAgg(node) => Some(format!(
+                "state tables: [{}]",
+                node.internal_tables
+                    .iter()
+                    .map(|tb| { self.add_table(tb) })
+                    .join(", ")
+            )),
+            stream_node::NodeBody::AppendOnlyTopN(node) => Some(format!(
+                "state table: {}",
+                self.add_table(node.get_table().unwrap())
+            )),
+            stream_node::NodeBody::HashJoin(node) => Some(format!(
+                "left table: {}, right table {},{}{}",
```

Comment:
When they are both `None`, the `,` will still be present. 😄

BTW, `left/right_degree_table` seems to be always `Some` currently?